File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        int need;
        for (int i=0; i<nums.size(); ++i){
            need = target - nums[i];
            if (auto it = mp.find(need); it != mp.end())
               return {mp[need], i};
            mp[nums[i]] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mp = {}
        for i,num in enumerate(nums):
            need = target - num
            if need in mp:
                return [mp[need],i]
            mp[num] = i
        return []
        